Source/Core/Core/Core.cpp

```diff
@@ -769,6 +770,48 @@ void RunAsCPUThread(std::function<void()> function)
     PauseAndLock(false, was_unpaused);
 }
 
+void RunOnCPUThread(std::function<void()> function, bool wait_for_completion)
+{
+  // If the CPU thread is not running, assume there is no active CPU thread we can race against.
+  if (!IsRunning() || IsCPUThread())
+  {
+    function();
+    return;
+  }
+
+  // Pause the CPU (set it to stepping mode).
+  const bool was_running = PauseAndLock(true, true);
+
+  // We create per-call events so that multiple threads can call this function concurrently.
```

Comment:
Isn't calling this from multiple threads impossible due to the reason stated in the header file?

Reply:
Indeed. The comment is left over from the older implementation.